priv/repo/migrations/20220418150726_add_publisher_to_projects_and_products.exs

```diff
@@ -0,0 +1,46 @@
+defmodule Oli.Repo.Migrations.AddPublisherToProjectsAndProducts do
+  use Ecto.Migration
+
+  import Ecto.Query, warn: false
+
+  alias Oli.Delivery.Sections.Section
+  alias Oli.Inventories
+  alias Oli.Inventories.Publisher
+  alias Oli.Repo
+
+  require Logger
+
+  def up do
+    default_publisher_attrs = %{name: Inventories.default_publisher_name(), email: "publisher@cmu.edu"}
+
+    case Inventories.find_or_create_publisher(default_publisher_attrs) do
+      {:ok, %Publisher{id: publisher_id}} ->
+        alter table(:sections) do
+          add :publisher_id, references(:publishers)
+        end
+
+        alter table(:projects) do
+          add :publisher_id, references(:publishers), null: false, default: publisher_id
+        end
+
+        flush()
+
+        #Set default publisher for products
+        from(s in Section, where: s.type == :blueprint)
+        |> Repo.update_all(set: [publisher_id: publisher_id])
+
+      {:error, error} ->
```

Comment:
Safe to assume this will stop the migration - 99% sure it will lol :)

Reply:
Decided to follow a different approach for the migration code so that it's independent from external modules. Also we're letting it fail when it has to now, so the migration isn't marked as successful in that case and it can be rerun